### ISSUE TITLE
[REVIEW]: PorousFlow: a multiphysics simulation code for coupled problems in porous media

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2176]
**Submitting author:** @WilkAndy (<a href="http://orcid.org/0000-0001-6472-9560">Andy Wilkins</a>)
**Repository:** <a href="https://github.com/idaholab/moose" target ="_blank">https://github.com/idaholab/moose</a>
**Version:** snapshot-20-10-27
**Editor:** @jedbrown
**Reviewer:** @jbrezmorf, @rpodgorney
**Archive:** <a href="https://doi.org/10.5281/zenodo.4071026" target="_blank">10.5281/zenodo.4071026</a>

## Status

[![status](https://joss.theoj.org/papers/0778dd9013e04c0ccde9fc6d2469e550/status.svg)](https://joss.theoj.org/papers/0778dd9013e04c0ccde9fc6d2469e550)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/0778dd9013e04c0ccde9fc6d2469e550"><img src="https://joss.theoj.org/papers/0778dd9013e04c0ccde9fc6d2469e550/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/0778dd9013e04c0ccde9fc6d2469e550/status.svg)](https://joss.theoj.org/papers/0778dd9013e04c0ccde9fc6d2469e550)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/idaholab/moose" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@jbrezmorf & @rpodgorney, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @jedbrown know.

✨ **Please try and complete your review in the next two weeks** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/idaholab/moose">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@WilkAndy) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/idaholab/moose">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@WilkAndy) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @jbrezmorf, @rpodgorney it looks like you're currently assigned to review this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02176/joss.02176/10.21105.joss.02176.pdf)

**Comment by jedbrown** [comment]
@jbrezmorf, @rpodgorney :wave: Welcome to JOSS and thanks for agreeing to review!

The comments from @whedon above outline the review process, which takes place in this thread (possibly with issues filed in the MOOSE repository, which is where PorousFlow resides). I'll be watching this thread if you have any questions.

The JOSS review is different from most other journals. Our goal is to work with the authors to help them meet our criteria instead of merely passing judgment on the submission. As such, the reviewers are encouraged to submit issues and pull requests on the software repository. When doing so, please mention https://github.com/openjournals/joss-reviews/issues/2176 so that a link is created to this thread (and I can keep an eye on what is happening). Please also feel free to comment and ask questions on this thread. In my experience, it is better to post comments/questions/suggestions as you come across them instead of waiting until you've reviewed the entire package.

We aim for reviews to be completed within a month or so. Please let me know if you require some more time. We can also use Whedon (our bot) to set automatic reminders if you know you'll be away for a known period of time.

Please feel free to ping me (@jedbrown) if you have any questions/concerns.

**Comment by jbrezmorf** [comment]
@jedbrown Hi Jed. Trying to proceed with the review I realized that my JOSS invitation has expired and I can not fill the checklist. Sorry, I didn't notice the invitation before.

**Comment by arfon** [comment]
@whedon re-invite @jbrezmorf as reviewer

**Comment by whedon** [comment]
OK, the reviewer has been re-invited.

@jbrezmorf please accept the invite by clicking this link: https://github.com/openjournals/joss-reviews/invitations

**Comment by arfon** [comment]
> @jedbrown Hi Jed. Trying to proceed with the review I realized that my JOSS invitation has expired and I can not fill the checklist. Sorry, I didn't notice the invitation before.

☝️ You should be re-invited now.

**Comment by jbrezmorf** [comment]
Software:
======= 
no major issues, some tips fo improvements in 
[the issue](https://github.com/idaholab/moose/issues/15268).

- fully coupled THM multiphase 
- use generic Newton method for solving the system of equations
- claimed implementation of all types of THM relationships
- supported by detailed documentation both for usage as for used numerical methods

Summary:
======= 
well written,  a typographic error in Acknowledgements: "Philipp Sch{”a}dle."

**Comment by arfon** [comment]
:wave: @rpodgorney - today we [reopened JOSS](https://blog.joss.theoj.org/2020/05/reopening-joss) for new submissions and are checking in on our existing reviews. Do you think you might be able to wrap up your review in the next 2-3 weeks?

**Comment by rpodgorney** [comment]
Sorry for letting this slip.  All in all no issues at all, the code, tests, and documentation are al superb.

The paper is well written and contains just the right amount of details without becoming overly verbose.

**Comment by rpodgorney** [comment]
@arfon this is my first review for JOSS.  I've looked through the code, documentation, and tests -- and feel it is OK to publish.  Other than the checklist, is there anything else I need to do to complete/document my approval?

**Comment by arfon** [comment]
> @arfon this is my first review for JOSS. I've looked through the code, documentation, and tests -- and feel it is OK to publish. Other than the checklist, is there anything else I need to do to complete/document my approval?

Nope, that's it. Over to @jedbrown to decide on next steps.

**Comment by jedbrown** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02176/joss.02176/10.21105.joss.02176.pdf)

**Comment by jedbrown** [comment]
Sorry about my delay. This is looking great.
1. There is a period missing after the Lime reference (end of paragraph).
2. "simply solve simplistic" -> "only solve simplistic" or some other phrasing.
3. Could you make any comments in the comparison with other software, perhaps in terms of capability, performance, and easy of use/extension?
4. You're welcome to include a figure if you'd like.
5. Would it make sense to write down the basic equations that are solved? Perhaps one or two lines for each of THMC just to get the structural gist across, or a figure representing the role and coupling. (It's sort of abrupt to launch directly into prior applications).
6. Can you double-check capitalization in your references? Our toolchain uses pandoc-citeproc, which is based on apa.csl. Its semantics are closer to biblatex than bibtex, which is required to be precisely APA-conforming. Anyway, there are extra braces needed, e.g., `{I}celand` or `{Iceland}` are needed.

**Comment by WilkAndy** [comment]
Thank you @jedbrown for all those comments.  We'll get onto them next week.

**Comment by jedbrown** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1007/s12665-012-1546-x is OK
- 10.1007/s00366-006-0049-3 is OK
- 10.1080/22020586.2019.12073198 is OK

MISSING DOIs

- https://doi.org/10.1016/j.softx.2020.100430 may be missing for title: MOOSE: Enabling Massively Parallel Multiphysics Simulation
- https://doi.org/10.21203/rs.2.17644/v1 may be missing for title: Impact of Effective Normal Stress on Capillary Pressure in a Single Natural Fracture

INVALID DOIs

- None
```

**Comment by WilkAndy** [comment]
Hi @jedbrown .   I've just submitted a PullRequest to MOOSE for the changes you request above.   Perhaps you'd like to review the changes there?   https://github.com/idaholab/moose/pull/15540

- Regarding the equations: I've included some, although we wanted this paper to be readable by a more general audience, so have included the equations towards the end of the paper.

- Regarding comparisons with other codes: I've included another paragraph that discusses some of the benefits of using MOOSE+PorousFlow. It's really hard to write this sort of thing without appearing to denigrate other very worthy codes!

- Done the DOIs mentioned above

**Comment by WilkAndy** [comment]
The above changes have now been merged into the `next` branch of the MOOSE repo.   Note the branch - at some stage we changed whedon's branch to `joss_porous_flow` , but that has not actually been used for all the editorial+reviewer changes.  So the `joss_porous_flow` branch should not be used for our JOSS paper: instead use the `next` branch.

Thanks!

andy

**Comment by jedbrown** [comment]
@whedon generate pdf from branch next

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch next. Reticulating splines etc...
```

**Comment by jedbrown** [comment]
@whedon check references

**Comment by whedon** [comment]
PDF failed to compile for issue #2176 with the following error: 

 sh: 1: cd: can't cd to tmp/2176
/app/vendor/ruby-2.4.4/lib/ruby/2.4.0/find.rb:43:in `block in find': No such file or directory - tmp/2176 (Errno::ENOENT)
	from /app/vendor/ruby-2.4.4/lib/ruby/2.4.0/find.rb:43:in `collect!'
	from /app/vendor/ruby-2.4.4/lib/ruby/2.4.0/find.rb:43:in `find'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-446a0298a33b/lib/whedon/processor.rb:61:in `find_paper_paths'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-446a0298a33b/bin/whedon:50:in `prepare'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-446a0298a33b/bin/whedon:119:in `<top (required)>'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `load'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `<main>'

**Comment by jedbrown** [comment]
@openjournals/dev I see the paper.md in the same place. Is this a Whedon issue?

**Comment by arfon** [comment]
@whedon generate pdf from branch next

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch next. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02176/joss.02176/10.21105.joss.02176.pdf)

**Comment by arfon** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1007/s12665-012-1546-x is OK
- 10.1007/s00366-006-0049-3 is OK
- 10.21203/rs.2.17644/v1 is OK
- 10.1080/22020586.2019.12073198 is OK

MISSING DOIs

- None

INVALID DOIs

- https://doi.org/10.1016/j.softx.2020.100430 is INVALID because of 'https://doi.org/' prefix
```

**Comment by arfon** [comment]
@jedbrown - not sure what was going on here. Seems to be working now.

**Comment by jedbrown** [comment]
Thanks @arfon

**Comment by WilkAndy** [comment]
I can fix the `https://doi.org/` early next week

**Comment by jedbrown** [comment]
While you're at it, could you avoid repeating the repository URL (which already exists in the sidebar)? Also, it's a bit unusual to launch into a bullet list of applications after only one short sentence saying what the package does. I suppose I don't object to the present organization, but an EiC might.

I have a question outside the context of the review: is the global conservation exact (to solver tolerance) or only up to discretization error? The skeleton velocity term not being in divergence form looks suspicious to me.

**Comment by WilkAndy** [comment]
MOOSE issue to address the "https" and "repo URL" at  https://github.com/idaholab/moose/issues/15591

**Comment by WilkAndy** [comment]
I've submitted a PullRequest to the MOOSE repo for these changes: https://github.com/idaholab/moose/pull/15592

**Comment by WilkAndy** [comment]
@whedon generate pdf from branch next

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch next
. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02176/joss.02176/10.21105.joss.02176.pdf)

**Comment by WilkAndy** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1007/s12665-012-1546-x is OK
- 10.1016/j.softx.2020.100430 is OK
- 10.1007/s00366-006-0049-3 is OK
- 10.21203/rs.2.17644/v1 is OK
- 10.1080/22020586.2019.12073198 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by WilkAndy** [comment]
Hi @jedbrown 

- i believe i've addressed your issues raised above
- i have checked the article proof - it looks fine
- the reference check is OK.

Good to publish?

andy

**Comment by WilkAndy** [comment]
Hi @jedbrown .  Just wondering about the status of this JOSS paper.   I think we've addressed all referee and editorial iteams and it's read to go.  If not, please tell me what needs to be done.

**Comment by jedbrown** [comment]
Sorry about my delays. It's looking great.
* Could you note that a Lagrangian mesh is used (so the mesh moves with `v_s`; this is done at the very start of the website docs, but didn't make it here yet)?
* Is it necessary to switch from operator to index notation for eq 2 and 3? (Perhaps yes if you don't like `:` for contraction over both indices, but I want to make sure it's an affirmative choice.)
* I think https://doi.org/10.1016/j.camwa.2020.05.014 is a preferred citation for OPM.
* "THMC" is only used twice and can just be spelled out.

Once you've addressed these, please:
- [ ] Make a tagged release of your software, and list the version tag of the archived version here.
- [ ] Archive the reviewed software in Zenodo or a similar DOI-granting service (e.g., figshare, an institutional repository)
- [ ] Check the archival deposit (e.g., in Zenodo) has the correct metadata. This includes the title (should match the paper title) and author list (make sure the list is correct and people who only made a small fix are not on it). You may also add the authors' ORCID.
- [ ] Please list the DOI of the archived version here.

I can then move forward with accepting the submission.

**Comment by WilkAndy** [comment]
Sounds good.  I shall make those changes and archive the software soon.  Idaho National Labs has been experiencing some sort of software/hardware/electrical problems for a while and consequently the MOOSE testing/merging/etc has been held up.  So it might be a couple of weeks before these changes+archiving gets completed.

**Comment by jedbrown** [comment]
FWIW, these small changes to the paper do not need to appear in the archived version, but it should be a tagged release that includes any substantive changes resulting from the review. I see [only one tag](https://github.com/idaholab/moose/tags) in the MOOSE repository; let us know if tagging a release (`v1.0.1` or some such) is problematic.

**Comment by kthyng** [comment]
Hi all! I am the rotating editor in chief this week. I see it has been awhile since there has been activity on this submission, but it looks pretty much done. @jedbrown what steps are left before you'd like me to take over to finish publishing? @WilkAndy have you finished the items listed just above here?

**Comment by WilkAndy** [comment]
Hi @kthyng , I just got back from holidays and am addressing a backlog of TODO items.  I have not implemented the items mentioned above by @jedbrown, but hope to do them this week.  However, because of a disastrous computer crash at Idaho National Labs, the whole MOOSE project has been severely impacted so I don't expect the changes, tagging, etc, will be reviewed and merged within the week, so I think you can safely forget about it during your editor-in-chief week.

**Comment by kthyng** [comment]
@WilkAndy Got it! No problem, happy to hear there is a path forward. Good luck with the computer crash.

**Comment by WilkAndy** [comment]
@jedbrown 's final suggestions have been implemented and merged into the master branch

tag = https://github.com/idaholab/moose/releases/tag/snapshot-20-10-27   (ie, tag name is snapshot-20-10-27 )

https://zenodo.org/record/4071026#.X5kq-9sRXOQ

Metadata is correct

doi = 10.5281/zenodo.4071026

**Comment by WilkAndy** [comment]
I think i've addressed all comments and this can be published

**Comment by WilkAndy** [comment]
Just wondering if i have to do anything else?

**Comment by WilkAndy** [comment]
Hi @kthyng .   I'm pinging you about this JOSS paper.  I'm pretty sure it's ready to publish, but @jedbrown isn't responding.  I think he might have accidentally unsubscribed to this issue.   Perhaps you can take it from here?

**Comment by jedbrown** [comment]
Sorry about the delay, I just got swamped.

**Comment by jedbrown** [comment]
@whedon set snapshot-20-10-27 as version

**Comment by whedon** [comment]
OK. snapshot-20-10-27 is the version.

**Comment by jedbrown** [comment]
@whedon generate pdf

**Comment by jedbrown** [comment]
@whedon set 10.5281/zenodo.4071026 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4071026" target="_blank">10.5281/zenodo.4071026</a> is the archive.

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02176/joss.02176/10.21105.joss.02176.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02176/joss.02176/10.21105.joss.02176.pdf) :page_facing_up: :point_left:

**Comment by jedbrown** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1941

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1941, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1007/s12665-012-1546-x is OK
- 10.1016/j.camwa.2020.05.014 is OK
- 10.1016/j.softx.2020.100430 is OK
- 10.1007/s00366-006-0049-3 is OK
- 10.21203/rs.2.17644/v1 is OK
- 10.1080/22020586.2019.12073198 is OK

MISSING DOIs

- 10.2139/ssrn.3365603 may be a valid DOI for title: Community Code for Simulating CO_2 Storage: Modelling Multiphase Flow with Coupled Geomechanics and Geochemistry Using the Open-Source Multiphysics Framework MOOSE

INVALID DOIs

- None
```

**Comment by jedbrown** [comment]
@WilkAndy It looks like the bib entry for that should be update. (It looks like the `21--26` is actually a date, so maybe you can check the rest of the formatting when you add the DOI.)

While you're at it, could you add the URL https://documentation.pflotran.org/ for PFLOTRAN (as recommended in the bibtex on their website)?

You can just put those small changes in your branch and we'll rebuild the PDF.

**Comment by WilkAndy** [comment]
Hi @jedbrown.   Thanks for all your work on this.  I've made the changes your requested above and the result can be found at

https://github.com/WilkAndy/moose    branch: joss_bib_16312

You may generate the PDF from there (I have checked it works) or you may wait for my PR (https://github.com/idaholab/moose/pull/16313) to make its way through the MOOSE process, which will 1day - 1week.

**Comment by jedbrown** [comment]
@whedon generate pdf from branch joss_bib_16312

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss_bib_16312. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02176/joss.02176/10.21105.joss.02176.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02176/joss.02176/10.21105.joss.02176.pdf) :page_facing_up: :point_left:

**Comment by WilkAndy** [comment]
I think the above compilation didn't use my repo .   I went to https://whedon.theoj.org/ and entered https://github.com/WilkAndy/moose/ and joss_bib_16312 , and the result is different (it contains the bib changes)

**Comment by jedbrown** [comment]
Hmm, I see this branch is in a different repository (https://github.com/WilkAndy/moose/tree/joss_bib_16312) and I don't know how to ask Whedon to build from that. But this paper should be ready for acceptance with either a manual build or once it lands in a branch of the main repository.

**Comment by arfon** [comment]
@whedon accept from branch joss_bib_16312

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1942

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1942, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true from branch joss_bib_16312 
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1007/s12665-012-1546-x is OK
- 10.1016/j.camwa.2020.05.014 is OK
- 10.1016/j.softx.2020.100430 is OK
- 10.1007/s00366-006-0049-3 is OK
- 10.21203/rs.2.17644/v1 is OK
- 10.1080/22020586.2019.12073198 is OK

MISSING DOIs

- 10.2139/ssrn.3365603 may be a valid DOI for title: Community Code for Simulating CO_2 Storage: Modelling Multiphase Flow with Coupled Geomechanics and Geochemistry Using the Open-Source Multiphysics Framework MOOSE

INVALID DOIs

- None
```

**Comment by arfon** [comment]
@jedbrown - is this (☝️) what you were wanting to do with the custom branch?

@WilkAndy - has the DOI suggested by Whedon above (`10.2139/ssrn.3365603`) been determined to be correct (or incorrect). If it looks correct for the reference item, please add it to your BibTeX file.

**Comment by WilkAndy** [comment]
Yep, @arfon, the DOI is correct, and i have added to paper.bib.  We are now waiting for the updates (https://github.com/idaholab/moose/pull/16313) to pass through the MOOSE review process and then we should be ready for publication.  Thank you everyone who contributed to this....

**Comment by arfon** [comment]
@whedon accept from branch joss_bib_16312

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1943

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1943, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true from branch joss_bib_16312 
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1007/s12665-012-1546-x is OK
- 10.1016/j.camwa.2020.05.014 is OK
- 10.1016/j.softx.2020.100430 is OK
- 10.1007/s00366-006-0049-3 is OK
- 10.21203/rs.2.17644/v1 is OK
- 10.1080/22020586.2019.12073198 is OK

MISSING DOIs

- 10.2139/ssrn.3365603 may be a valid DOI for title: Community Code for Simulating CO_2 Storage: Modelling Multiphase Flow with Coupled Geomechanics and Geochemistry Using the Open-Source Multiphysics Framework MOOSE

INVALID DOIs

- None
```

**Comment by WilkAndy** [comment]
No, the above is still the old version.  It is not from https://github.com/WilkAndy/moose/tree/joss_bib_16312 .    I'm happy to wait till the bib changes make their way through the review process at https://github.com/idaholab/moose/pull/16313

**Comment by arfon** [comment]
Ah, OK. I think I know what's going on here - @whedon doesn't know how to build from a fork branch so yes, we'll need to wait for these changes (https://github.com/idaholab/moose/pull/16313) to land in `https://github.com/idaholab/moose`.

**Comment by jedbrown** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02176/joss.02176/10.21105.joss.02176.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02176/joss.02176/10.21105.joss.02176.pdf) :page_facing_up: :point_left:

**Comment by arfon** [comment]
@whedon accept from branch next

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1007/s12665-012-1546-x is OK
- 10.1016/j.camwa.2020.05.014 is OK
- 10.1016/j.softx.2020.100430 is OK
- 10.1007/s00366-006-0049-3 is OK
- 10.2139/ssrn.3365603 is OK
- 10.21203/rs.2.17644/v1 is OK
- 10.1080/22020586.2019.12073198 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1944

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1944, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true from branch next 
```

**Comment by arfon** [comment]
@jedbrown - this looks in good shape to publish now, would you agree?

**Comment by jedbrown** [comment]
Looks good to me. Thanks, @arfon.

**Comment by WilkAndy** [comment]
I concur - the paper looks as it should.  Thank you everyone who was involved in the review process

**Comment by WilkAndy** [comment]
@whedon accept deposit=true from branch next

**Comment by whedon** [comment]
I'm sorry @WilkAndy, I'm afraid I can't do that. That's something only editor-in-chiefs are allowed to do.

**Comment by arfon** [comment]
@whedon accept deposit=true from branch next

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1333149777719140359) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/1945
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.02176](https://doi.org/10.21105/joss.02176)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by arfon** [comment]
@jbrezmorf, @rpodgorney - many thanks for your reviews here and to @jedbrown for editing ✨

@WilkAndy - your paper is now accepted into JOSS :zap::rocket::boom:

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.02176/status.svg)](https://doi.org/10.21105/joss.02176)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.02176">
  <img src="https://joss.theoj.org/papers/10.21105/joss.02176/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.02176/status.svg
   :target: https://doi.org/10.21105/joss.02176
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.02176/status.svg)](https://doi.org/10.21105/joss.02176)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
